packages/ui/src/components/Select/hooks/useItems.tsx

```diff
@@ -0,0 +1,56 @@
+import get from 'lodash/get'
+import React, { useMemo } from 'react'
+
+import { Typography } from '../../Typography'
+import { OptGroup, Option } from '../select'
+
+import type { OptionsItem, OptionsGroupType, OptionType } from '../select'
+
+type UseItems = (items?: OptionsItem[]) => null | (null | JSX.Element)[]
+
+const convertOptions = (items: OptionsItem[]) => {
+    if (items.length < 1) return null
+
+    return items
+        .map((item,  index) => {
+            if (item === null) return null
+
+            const { key } = item
+            const mergedKey = key || index
+            const isOptionsGroup = Boolean(get(item, 'options', false))
+
+            if (isOptionsGroup) {
+                // Options group
+                const { label, options } = item as OptionsGroupType
```

Comment:
why typecasting here and below?

Reply:
Without this, an error appears with the message `Property 'textType' does not exist on type 'OptionType | OptionsGroupType'`

Reply:
fixed